### PR TITLE
feat(kafka): support serializing producer

### PR DESCRIPTION
**Pull request by piochelepiotr**
The confluent_kafka library has a serializing producer / deserializing consumer.
This PR adds support to it.
Right now, both are experimental, but they have been in the library for over a year, so we had some customers on data streams using it.

I added tests, and I also tested with a small local application.

## Checklist

- [x] Change(s) are motivated and described in the PR description.
- [x] Testing strategy is described if automated tests are not included in the PR.
- [x] Risk is outlined (performance impact, potential for breakage, maintainability, etc).
- [x] Change is maintainable (easy to change, telemetry, documentation).
- [x] [Library release note guidelines](https://ddtrace.readthedocs.io/en/stable/releasenotes.html) are followed. If no release note is required, add label `changelog/no-changelog`.
- [x] Documentation is included (in-code, generated user docs, [public corp docs](https://github.com/DataDog/documentation/)).
- [x] Backport labels are set (if [applicable](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting))

## Reviewer Checklist

- [x] Title is accurate.
- [x] No unnecessary changes are introduced.
- [x] Description motivates each change.
- [x] Avoids breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes unless absolutely necessary.
- [x] Testing strategy adequately addresses listed risk(s).
- [x] Change is maintainable (easy to change, telemetry, documentation).
- [x] Release note makes sense to a user of the library.
- [x] Reviewer has explicitly acknowledged and discussed the performance implications of this PR as reported in the benchmarks PR comment.
- [x] Backport labels are set in a manner that is consistent with the [release branch maintenance policy](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting)

### CODE REVIEW

**Review by majorgreys — DISMISSED** [review]

File: ddtrace/contrib/kafka/patch.py [hunk 1]
```diff
@@ -77,6 +81,8 @@ def patch():
 
     confluent_kafka.Producer = TracedProducer
     confluent_kafka.Consumer = TracedConsumer
+    confluent_kafka.SerializingProducer = TracedProducer
+    confluent_kafka.DeserializingConsumer = TracedConsumer
```
Comment (Yun-Kim) [inline ddtrace/contrib/kafka/patch.py:85]: Can we add the same safety check to see if `SerializingProducer/DeserializingConsumer` exists before we patch it?

File: releasenotes/notes/kafka-support-serializing-producer-fb01f5288ae116f4.yaml [hunk 2]
```diff
@@ -0,0 +1,3 @@
+features:
+  - |
+    kafka: Add support for the Kafka serializing producer and deserializing consumer.
```
Comment (Yun-Kim) [inline releasenotes/notes/kafka-support-serializing-producer-fb01f5288ae116f4.yaml:3]:
```suggestion
    kafka: Adds support for the Kafka serializing producer and deserializing consumer.
```

File: ddtrace/contrib/kafka/patch.py (hunk 1 above)
Comment (piochelepiotr) [reply ddtrace/contrib/kafka/patch.py]: Makes sense. Updated.

File: releasenotes/notes/kafka-support-serializing-producer-fb01f5288ae116f4.yaml [hunk 3]
```diff
@@ -1,3 +1,3 @@
 features:
   - |
-    kafka: Add support for the Kafka serializing producer and deserializing consumer.
+    kafka: Adds support for the Kafka serializing producer and de serializing consumer.
```
Comment (Yun-Kim) [inline releasenotes/notes/kafka-support-serializing-producer-fb01f5288ae116f4.yaml:3]: We can keep `deserializing` and just add it to the `spelling_wordlist.txt` file, like so: https://github.com/DataDog/dd-trace-py/blob/1.x/docs/spelling_wordlist.txt

**Review by Yun-Kim — COMMENTED** [review]
Just one small nit but LGTM!

**Review by Yun-Kim — APPROVED** [review]

**Pull request merged**